modules/carto/src/layers/utils.ts

```diff
@@ -16,13 +16,26 @@ export function injectAccessToken(loadOptions: any, accessToken: string): void {
   }
 }
 
-export function mergeBoundaryData(geometry: VectorTile, properties: PropertiesTile): VectorTile {
+export function mergeBoundaryData(
+  geometry: VectorTile,
+  properties: PropertiesTile,
+  matchingColumn: string
+): VectorTile {
   const mapping = {};
-  for (const {geoid, ...rest} of properties.properties) {
-    if (geoid in mapping) {
-      log.warn('Duplicate geoid key in boundary mapping, using first occurance')();
+  for (const property of properties.properties) {
+    const matchingKey = property[matchingColumn];
+    if (!matchingKey || matchingKey === '') {
+      log.warn(
+        `Properties: Missing value for matchingKey with matchingColumn: ${matchingColumn}`
+      )();
+    }
+    const row = {...property};
+    delete row[matchingColumn];
```

Comment:
What would happen if we didn't delete the matching column? Just wondering if we can avoid copying every row, and/or should leave a comment so we don't try to make that optimization incorrectly later.

Reply:
Nothing much, we would just expose the `matchingColumn` to the layer accessors. I guess that doesn't hurt and might in fact be useful, e.g. visualizing the ZIP code